src/app/modules/shopping-cart/components/containers/shopping-cart-details/shopping-cart-details.component.ts

```diff
@@ -1,12 +1,57 @@
-import { Component } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Component, OnInit } from '@angular/core';
+import { UntilDestroy } from '@ngneat/until-destroy';
+import { Observable, of } from 'rxjs';
 import { productsMockList } from 'src/app/components/mocks/products.mock';
-import { Product } from 'src/app/modules/shared/types/product';
+import { ShoppingCartService } from 'src/app/modules/shared/services/shopping.cart.service';
+import { OrderProductWithId } from 'src/app/modules/shared/types/order.product';
+import { OrderProduct } from 'src/app/modules/shared/types/product.order';
+import { OrderByCustomerDto } from 'src/app/modules/shared/types/product.order.with.customer';
+import { environment } from 'src/environments/environment';
 
+@UntilDestroy({
+  checkProperties: true,
+})
 @Component({
   selector: 'app-shopping-cart-details',
   templateUrl: './shopping-cart-details.component.html',
   styleUrls: ['./shopping-cart-details.component.scss'],
 })
 export class ShoppingCartDetailsComponent {
-  productList: Product[] = productsMockList;
+  private shoppingCartService: ShoppingCartService;
+  productList: OrderProduct[];
+  constructor(
+    shoppingCartService: ShoppingCartService,
+    private http: HttpClient
+  ) {
+    this.shoppingCartService = shoppingCartService;
+    this.productList = this.shoppingCartService.getShoppingCart();
+  }
+
+  getProducts(): Observable<OrderProduct[]> {
+    return of((this.productList = this.shoppingCartService.getShoppingCart()));
+  }
+  catchProductDelete(productId: string) {
+    this.productList = this.shoppingCartService.removeFromShoppingCart(
+      this.productList,
+      productId
+    );
+  }
+  catchCheckoutOrder(productList: OrderProduct[]) {
+    let dummyCustomer = 'bacb89d7-9070-426f-a662-facb538c8da9';
```

Comment:
Try to get used to use ```const``` when you don't need to modify values. For example, here the hardcoded value of the interviewer will not change, so you could use const. Available also for the object declared below